File: 351004/Barsukov/app/services/note_service.py
from app.repository.memory import note_repo, issue_repo
from app.schemas.note import NoteRequestTo, NoteResponseTo
from app.core.exceptions import AppException


class NoteService:
    def create(self, dto: NoteRequestTo) -> NoteResponseTo:
        # Проверяем, что issue существует
        if not issue_repo.find_by_id(dto.issueId):
            raise AppException(400, "Issue not found", 12)
        return NoteResponseTo(**note_repo.save(dto.model_dump()))

    def get_all(self):
        return [NoteResponseTo(**n) for n in note_repo.find_all()]

    def get_by_id(self, id: int):
        res = note_repo.find_by_id(id)
        if not res:
            raise AppException(404, "Note not found", 13)
        return NoteResponseTo(**res)

    def update(self, id: int, dto: NoteRequestTo) -> NoteResponseTo:
        # Проверяем, что note существует
        if not note_repo.find_by_id(id):
            raise AppException(404, "Note not found", 14)
        # Проверяем, что issue существует
        if not issue_repo.find_by_id(dto.issueId):
            raise AppException(400, "Issue not found", 15)
        data = dto.model_dump()
        data["id"] = id
        saved = note_repo.save(data)
        return NoteResponseTo(**saved)

    def delete(self, id: int):
        if not note_repo.delete(id):
            raise AppException(404, "Note not found", 16)

File: 351004/Beliaev/restapp/src/main/java/com/example/restapp/service/NoteService.java
package com.example.restapp.service;

import com.example.restapp.dto.request.NoteRequestTo;
import com.example.restapp.dto.response.NoteResponseTo;
import com.example.restapp.exception.EntityNotFoundException;
import com.example.restapp.mapper.NoteMapper;
import com.example.restapp.model.Note;
import com.example.restapp.repository.NoteRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
public class NoteService {
    private final NoteRepository repository;
    private final NoteMapper mapper;

    public NoteResponseTo create(NoteRequestTo request) {
        Note note = mapper.toEntity(request);
        Note saved = repository.save(note);
        return mapper.toResponse(saved);
    }

    public List<NoteResponseTo> getAll() {
        return repository.findAll().stream()
                .map(mapper::toResponse)
                .collect(Collectors.toList());
    }

    public NoteResponseTo getById(Long id) {
        return repository.findById(id)
                .map(mapper::toResponse)
                .orElseThrow(() -> new EntityNotFoundException("Note not found with id: " + id));
    }

    public NoteResponseTo update(Long id, NoteRequestTo request) {
        Note note = repository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException("Note not found with id: " + id));
        mapper.updateEntityFromDto(request, note);
        repository.update(note);
        return mapper.toResponse(note);
    }

    public void delete(Long id) {
        if (!repository.deleteById(id)) {
            throw new EntityNotFoundException("Note not found with id: " + id);
        }
    }
}